### PR TITLE
docs: add docs for using the alchemy signer

### PR DESCRIPTION
**Pull request by moldy530**
# Pull Request Checklist

- [ ] Did you add new tests and confirm existing tests pass? (`yarn test`)
- [ ] Did you update relevant docs? (docs are found in the `site` folder, and guidelines for updating/adding docs can be found in the [contribution guide](https://github.com/alchemyplatform/aa-sdk/blob/main/CONTRIBUTING.md))
- [ ] Do your commits follow the [Conventional Commits](https://www.conventionalcommits.org/en/v1.0.0/) standard?
- [ ] Does your PR title also follow the [Conventional Commits](https://www.conventionalcommits.org/en/v1.0.0/) standard?
- [ ] If you have a breaking change, is it [correctly reflected in your commit message](https://www.conventionalcommits.org/en/v1.0.0/#examples)? (e.g. `feat!: breaking change`)
- [ ] Did you run lint (`yarn lint:check`) and fix any issues? (`yarn lint:write`)
- [ ] Did you follow the [contribution guidelines](https://github.com/alchemyplatform/aa-sdk/blob/main/CONTRIBUTING.md)?

---

### Detailed summary
- Added a new page "Alchemy Signer" to the sidebar with multiple sub-pages.
- Added a new file `usePromise.ts` in the `alchemy` directory.
- Updated the `signer.ts` file in the `alchemy` directory with new imports and configurations.
- Removed the `SessionManagerParamsSchema` import from the `client/index.ts` file.
- Updated the `sidebar` file in the `packages/aa-alchemy.ts` directory with new links.
- Added new markdown files for the `disconnect`, `getAddress`, `getUser`, `signMessage`, `signTypedData`, `getAuthDetails`, `addPasskey`, and `exportWallet` methods in the `signer` directory.

> The following files were skipped due to too many changes: `site/packages/aa-alchemy/signer/exportWallet.md`, `site/packages/aa-alchemy/signer/authenticate.md`, `site/snippets/signers/alchemy/SignupLoginComponent.tsx`, `site/packages/aa-alchemy/signer/overview.md`, `site/signers/alchemy-signer.md`

> ✨ Ask PR-Codex anything about this PR by commenting with `/codex {your question}`

### CODE REVIEW

File: packages/alchemy/src/signer/client/index.ts [hunk 1]
```diff
@@ -26,7 +25,6 @@ export const AlchemySignerClientParamsSchema = z.object({
     iframeElementId: z.string().default("turnkey-iframe"),
     iframeContainerId: z.string(),
   }),
-  sessionConfig: SessionManagerParamsSchema.optional(),
```
Comment (moldy530) [inline packages/alchemy/src/signer/client/index.ts:29]: this is not used on the signer anymore

**Comment by denniswon** [comment]
I don't think we have ever added aa-signers doc right? should we include the generic aa-signers doc along with this as well? cc: @avasisht23

**Comment by denniswon** [comment]
Looking at this PR

I really feel like AlchemySigner should live in a separate repo and doc. By putting in AlchemySigners into aa-sdk, now we are introducing components that are not much relevant to other aa-sdk packages or purpose (to interact with SmartContractAccount). aa-alchemy is different because it is about using Alchemy to interact with SmartContractAccount so the code and docs are aligned with other aa-sdk packages in terms of functionality and purpose, but the signer code/doc quite out of the entire story of aa-sdk.

cc: @rthomare @moldy530 @avasisht23 @willhennessy

**Review by denniswon — DISMISSED** [review]

File: site/packages/aa-alchemy/signer/authenticate.md [hunk 2]
```diff
@@ -0,0 +1,69 @@
...
+
+const bundlePromise = new Promise(async (resolve) => {
+  const otpFromUser = await getOtpFromUser();
+  resolve(otpFromUser);
+});
+
+const user = await signer.authenticate({
+  type: "email",
+  email: "user@email.com",
+  bundle: bundlePromise,
```
Comment (denniswon) [inline site/packages/aa-alchemy/signer/authenticate.md:40]: what is a bundle exactly?

File: site/packages/aa-alchemy/signer/authenticate.md [hunk 3]
```diff
@@ -0,0 +1,69 @@
...
+
+## Usage
+
+::: code-group
+
+```ts
+import { signer } from "./signer";
+
+const bundlePromise = new Promise(async (resolve) => {
+  const otpFromUser = await getOtpFromUser();
```
Comment (denniswon) [inline site/packages/aa-alchemy/signer/authenticate.md:33]: put an inline comment explaining that `getOtpFromUser` is something that user will develop.

File: site/packages/aa-alchemy/signer/authenticate.md [hunk 4]
```diff
@@ -0,0 +1,69 @@
...
+
+`Promise<User>` -- on success returns a `User` object representing the authenticated user.
+
+## Parameters
+
+`AuthParams` -- an object that contains the following properties:
+
+```ts
+export type AuthParams =
+  | { type: "email"; email: string; bundle: Promise<string> }
```
Comment (denniswon) [inline site/packages/aa-alchemy/signer/authenticate.md:58]: we should put a clear explanation of what bundle is here and what it is doing/purpose.

File: site/packages/aa-alchemy/signer/authenticate.md (hunk 2 above)
Comment (moldy530) [reply site/packages/aa-alchemy/signer/authenticate.md:40]: bundle is the OTP string that a user is emailed. we'll be killing this in the next week or two anyways so likely won't add much more than that for description

**Comment by moldy530** [comment]
@denniswon added a comment about the OTP. we'll be killing this concept before beta launch

**Review by denniswon — APPROVED** [review]

**Pull request merged**